From 26f146964c13d8043ef027bf059befacdd1cd148 Mon Sep 17 00:00:00 2001
From: Christophe Roullier <christophe.roullier@st.com>
Date: Fri, 16 Apr 2021 16:31:08 +0200
Subject: [PATCH 382/529] ARM: dts: stm32: Add support to retrieve Ethernet
 @MAC with nvmem on stm32mp13

MAC1 and MAC2 are fused on OTP (57/58), two mechanisms to retrieve them:
- use "aliases": bootloader retrieve it and set "local-mac-address",
which is used during probe to set MAC address
- if "aliases" not set for ethernet, nvmem read directly OTP fused at
probe to set MAC address

Signed-off-by: Christophe Roullier <christophe.roullier@st.com>
Change-Id: Iad564edd07682d7ad6e7ef0e529987180d22c14e
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/229830
Reviewed-by: Eric FOURMONT <eric.fourmont-ext@st.com>
Reviewed-by: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Tested-by: Eric FOURMONT <eric.fourmont-ext@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/284374
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Alexandre TORGUE <alexandre.torgue@foss.st.com>
---
 arch/arm/boot/dts/stm32mp131.dtsi    | 6 ++++++
 arch/arm/boot/dts/stm32mp135f-dk.dts | 4 ++++
 2 files changed, 10 insertions(+)

--- a/arch/arm/boot/dts/stm32mp131.dtsi
+++ b/arch/arm/boot/dts/stm32mp131.dtsi
@@ -1582,6 +1582,12 @@
 			ts_cal2: calib@5e {
 				reg = <0x5e 0x2>;
 			};
+			ethernet_mac1_address: mac1@e4 {
+				reg = <0xe4 0x6>;
+			};
+			ethernet_mac2_address: mac2@ea {
+				reg = <0xea 0x6>;
+			};
 		};
 
 		/*
--- a/arch/arm/boot/dts/stm32mp135f-dk.dts
+++ b/arch/arm/boot/dts/stm32mp135f-dk.dts
@@ -210,6 +210,8 @@
 	phy-mode = "rmii";
 	max-speed = <100>;
 	phy-handle = <&phy0_eth1>;
+	nvmem-cells = <&ethernet_mac1_address>;
+	nvmem-cell-names = "mac-address";
 
 	mdio1 {
 		#address-cells = <1>;
@@ -235,6 +237,8 @@
 	phy-handle = <&phy0_eth2>;
 	st,ext-phyclk;
 	phy-supply = <&scmi_v3v3_sw>;
+	nvmem-cells = <&ethernet_mac2_address>;
+	nvmem-cell-names = "mac-address";
 
 	mdio1 {
 		#address-cells = <1>;
